127.0.0.1 - - [17/Mar/2025 01:19:44] "GET /example1.html HTTP/1.1" 200 -
127.0.0.1 - - [17/Mar/2025 01:19:44] "GET /_static/documentation_options.js HTTP/1.1" 200 -
127.0.0.1 - - [17/Mar/2025 01:19:44] "GET /_static/pygments.css HTTP/1.1" 200 -
127.0.0.1 - - [17/Mar/2025 01:19:44] "GET /_static/basic.css HTTP/1.1" 200 -
127.0.0.1 - - [17/Mar/2025 01:19:44] "GET /_static/language_data.js HTTP/1.1" 200 -
127.0.0.1 - - [17/Mar/2025 01:19:50] "GET /_static/bootstrap-sphinx.css HTTP/1.1" 200 -
----------------------------------------
Exception happened during processing of request from ('127.0.0.1', 53016)
Traceback (most recent call last):
  File "/opt/anaconda3/envs/myenv/lib/python3.8/socketserver.py", line 316, in _handle_request_noblock
    self.process_request(request, client_address)
  File "/opt/anaconda3/envs/myenv/lib/python3.8/socketserver.py", line 347, in process_request
    self.finish_request(request, client_address)
  File "/opt/anaconda3/envs/myenv/lib/python3.8/socketserver.py", line 360, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/opt/anaconda3/envs/myenv/lib/python3.8/http/server.py", line 664, in __init__
    super().__init__(*args, **kwargs)
  File "/opt/anaconda3/envs/myenv/lib/python3.8/socketserver.py", line 747, in __init__
    self.handle()
  File "/opt/anaconda3/envs/myenv/lib/python3.8/http/server.py", line 435, in handle
    self.handle_one_request()
  File "/opt/anaconda3/envs/myenv/lib/python3.8/http/server.py", line 423, in handle_one_request
    method()
  File "/opt/anaconda3/envs/myenv/lib/python3.8/http/server.py", line 671, in do_GET
    self.copyfile(f, self.wfile)
  File "/opt/anaconda3/envs/myenv/lib/python3.8/http/server.py", line 870, in copyfile
    shutil.copyfileobj(source, outputfile)
  File "/opt/anaconda3/envs/myenv/lib/python3.8/shutil.py", line 208, in copyfileobj
    fdst_write(buf)
  File "/opt/anaconda3/envs/myenv/lib/python3.8/socketserver.py", line 826, in write
    self._sock.sendall(b)
BrokenPipeError: [Errno 32] Broken pipe
----------------------------------------
127.0.0.1 - - [17/Mar/2025 01:19:50] "GET /example1.html HTTP/1.1" 200 -
127.0.0.1 - - [17/Mar/2025 01:19:50] "GET /_static/bootstrap-sphinx.css HTTP/1.1" 200 -
127.0.0.1 - - [17/Mar/2025 01:19:50] "GET /_static/documentation_options.js HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:19:50] "GET /_static/language_data.js HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:19:50] "GET /_static/basic.css HTTP/1.1" 200 -
127.0.0.1 - - [17/Mar/2025 01:19:50] "GET /_static/pygments.css HTTP/1.1" 200 -
127.0.0.1 - - [17/Mar/2025 01:19:50] code 404, message File not found
127.0.0.1 - - [17/Mar/2025 01:19:50] "GET /runestone/static/favicon.ico HTTP/1.1" 404 -
127.0.0.1 - - [17/Mar/2025 01:20:01] "GET /plan2.html HTTP/1.1" 200 -
127.0.0.1 - - [17/Mar/2025 01:20:01] "GET /_static/documentation_options.js HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:20:01] "GET /_static/language_data.js HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:20:01] "GET /_static/pygments.css HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:20:01] "GET /_static/basic.css HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:20:01] "GET /_static/bootstrap-sphinx.css HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:20:01] "GET /_images/bbq_URL.gif HTTP/1.1" 200 -
127.0.0.1 - - [17/Mar/2025 01:20:02] "GET /plan3.html HTTP/1.1" 200 -
127.0.0.1 - - [17/Mar/2025 01:20:10] "GET /index.html HTTP/1.1" 200 -
127.0.0.1 - - [17/Mar/2025 01:20:10] "GET /_static/documentation_options.js HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:20:10] "GET /_static/basic.css HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:20:10] "GET /_static/language_data.js HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:20:10] "GET /_static/pygments.css HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:20:16] "GET /_static/bootstrap-sphinx.css HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:20:16] code 404, message File not found
127.0.0.1 - - [17/Mar/2025 01:20:16] "GET /runestone/static/favicon.ico HTTP/1.1" 404 -
127.0.0.1 - - [17/Mar/2025 01:20:16] "GET /example1.html HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:20:31] "GET /activity-writing-1.html HTTP/1.1" 200 -
127.0.0.1 - - [17/Mar/2025 01:20:31] "GET /_static/documentation_options.js HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:20:31] "GET /_static/language_data.js HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:20:31] "GET /_static/basic.css HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:20:31] "GET /_static/pygments.css HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:20:31] "GET /_static/bootstrap-sphinx.css HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:20:31] "GET /_images/rate_my_prof_herman.png HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:20:31] "GET /_images/rate_my_prof_alawini.png HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:20:31] "GET /_images/rate_my_prof_herman_tags.png HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:20:31] code 404, message File not found
127.0.0.1 - - [17/Mar/2025 01:20:31] "GET /runestone/static/favicon.ico HTTP/1.1" 404 -
----------------------------------------
Exception happened during processing of request from ('127.0.0.1', 53063)
Traceback (most recent call last):
  File "/opt/anaconda3/envs/myenv/lib/python3.8/http/server.py", line 722, in send_head
    f = open(path, 'rb')
FileNotFoundError: [Errno 2] No such file or directory: '/Users/marailith/VisualStudioCode/cs-stars/pfp-ebooks/build/PurposeFirstWebScraping/runestone/static/favicon.ico'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/anaconda3/envs/myenv/lib/python3.8/socketserver.py", line 316, in _handle_request_noblock
    self.process_request(request, client_address)
  File "/opt/anaconda3/envs/myenv/lib/python3.8/socketserver.py", line 347, in process_request
    self.finish_request(request, client_address)
  File "/opt/anaconda3/envs/myenv/lib/python3.8/socketserver.py", line 360, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/opt/anaconda3/envs/myenv/lib/python3.8/http/server.py", line 664, in __init__
    super().__init__(*args, **kwargs)
  File "/opt/anaconda3/envs/myenv/lib/python3.8/socketserver.py", line 747, in __init__
    self.handle()
  File "/opt/anaconda3/envs/myenv/lib/python3.8/http/server.py", line 435, in handle
    self.handle_one_request()
  File "/opt/anaconda3/envs/myenv/lib/python3.8/http/server.py", line 423, in handle_one_request
    method()
  File "/opt/anaconda3/envs/myenv/lib/python3.8/http/server.py", line 668, in do_GET
    f = self.send_head()
  File "/opt/anaconda3/envs/myenv/lib/python3.8/http/server.py", line 724, in send_head
    self.send_error(HTTPStatus.NOT_FOUND, "File not found")
  File "/opt/anaconda3/envs/myenv/lib/python3.8/http/server.py", line 490, in send_error
    self.wfile.write(body)
  File "/opt/anaconda3/envs/myenv/lib/python3.8/socketserver.py", line 826, in write
    self._sock.sendall(b)
BrokenPipeError: [Errno 32] Broken pipe
----------------------------------------
127.0.0.1 - - [17/Mar/2025 01:20:32] "GET /activity-writing-2.html HTTP/1.1" 200 -
127.0.0.1 - - [17/Mar/2025 01:20:37] "GET /activity-writing-2.html HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:20:38] "GET /_static/documentation_options.js HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:20:38] "GET /_static/basic.css HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:20:38] "GET /_static/pygments.css HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:20:38] "GET /_static/bootstrap-sphinx.css HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:20:38] "GET /_static/language_data.js HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:20:38] code 404, message File not found
127.0.0.1 - - [17/Mar/2025 01:20:38] "GET /runestone/static/favicon.ico HTTP/1.1" 404 -
127.0.0.1 - - [17/Mar/2025 01:20:38] "GET /activity-writing-3.html HTTP/1.1" 200 -
127.0.0.1 - - [17/Mar/2025 01:20:42] "GET /activity-debugging.html HTTP/1.1" 200 -
----------------------------------------
Exception happened during processing of request from ('127.0.0.1', 53080)
Traceback (most recent call last):
  File "/opt/anaconda3/envs/myenv/lib/python3.8/socketserver.py", line 316, in _handle_request_noblock
    self.process_request(request, client_address)
  File "/opt/anaconda3/envs/myenv/lib/python3.8/socketserver.py", line 347, in process_request
    self.finish_request(request, client_address)
  File "/opt/anaconda3/envs/myenv/lib/python3.8/socketserver.py", line 360, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/opt/anaconda3/envs/myenv/lib/python3.8/http/server.py", line 664, in __init__
    super().__init__(*args, **kwargs)
  File "/opt/anaconda3/envs/myenv/lib/python3.8/socketserver.py", line 747, in __init__
    self.handle()
  File "/opt/anaconda3/envs/myenv/lib/python3.8/http/server.py", line 435, in handle
    self.handle_one_request()
  File "/opt/anaconda3/envs/myenv/lib/python3.8/http/server.py", line 423, in handle_one_request
    method()
  File "/opt/anaconda3/envs/myenv/lib/python3.8/http/server.py", line 671, in do_GET
    self.copyfile(f, self.wfile)
  File "/opt/anaconda3/envs/myenv/lib/python3.8/http/server.py", line 870, in copyfile
    shutil.copyfileobj(source, outputfile)
  File "/opt/anaconda3/envs/myenv/lib/python3.8/shutil.py", line 208, in copyfileobj
    fdst_write(buf)
  File "/opt/anaconda3/envs/myenv/lib/python3.8/socketserver.py", line 826, in write
    self._sock.sendall(b)
BrokenPipeError: [Errno 32] Broken pipe
----------------------------------------
127.0.0.1 - - [17/Mar/2025 01:20:42] code 404, message File not found
127.0.0.1 - - [17/Mar/2025 01:20:42] "GET /favicon.ico HTTP/1.1" 404 -
127.0.0.1 - - [17/Mar/2025 01:20:50] "GET /activity-debugging.html HTTP/1.1" 200 -
127.0.0.1 - - [17/Mar/2025 01:20:50] "GET /_static/documentation_options.js HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:20:50] "GET /_static/pygments.css HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:20:50] "GET /_static/basic.css HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:20:50] "GET /_static/bootstrap-sphinx.css HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:20:50] "GET /_static/language_data.js HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:20:50] "GET /_images/frodo_adoption.png HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:20:50] "GET /activity-explaining.html HTTP/1.1" 200 -
127.0.0.1 - - [17/Mar/2025 01:21:55] "GET /index.html HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:21:55] "GET /_static/basic.css HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:21:55] "GET /_static/pygments.css HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:21:55] "GET /_static/bootstrap-sphinx.css HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:21:55] "GET /_static/documentation_options.js HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:21:55] "GET /_static/jquery-fix.js?v=1226591B HTTP/1.1" 200 -
127.0.0.1 - - [17/Mar/2025 01:21:55] "GET /_static/theme.js?v=D899FB97 HTTP/1.1" 200 -
127.0.0.1 - - [17/Mar/2025 01:21:55] "GET /_static/language_data.js HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:21:55] code 404, message File not found
127.0.0.1 - - [17/Mar/2025 01:21:55] "GET /runestone/static/favicon.ico HTTP/1.1" 404 -
127.0.0.1 - - [17/Mar/2025 01:21:56] "GET /example1.html HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:21:58] "GET /activity-writing-2.html HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:21:58] code 404, message File not found
127.0.0.1 - - [17/Mar/2025 01:21:58] "GET /runestone/static/favicon.ico HTTP/1.1" 404 -
127.0.0.1 - - [17/Mar/2025 01:21:58] "GET /activity-writing-3.html HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:22:03] "GET /activity-explaining.html HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:22:03] "GET /_static/html4css1.css?v=4533ACD3 HTTP/1.1" 200 -
127.0.0.1 - - [17/Mar/2025 01:22:03] "GET /_static/jquery-fix.js?v=1226591B HTTP/1.1" 200 -
127.0.0.1 - - [17/Mar/2025 01:22:03] "GET /_static/theme.js?v=D899FB97 HTTP/1.1" 200 -
127.0.0.1 - - [17/Mar/2025 01:22:03] "GET /_images/news_cunningham_multiple.png HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:22:03] code 404, message File not found
127.0.0.1 - - [17/Mar/2025 01:22:03] "GET /runestone/static/favicon.ico HTTP/1.1" 404 -
127.0.0.1 - - [17/Mar/2025 01:22:03] "GET /downloads.html HTTP/1.1" 200 -
127.0.0.1 - - [17/Mar/2025 01:22:06] "GET /activity-debugging.html HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:22:06] "GET /_static/jquery-fix.js?v=1226591B HTTP/1.1" 200 -
127.0.0.1 - - [17/Mar/2025 01:22:06] "GET /_static/theme.js?v=D899FB97 HTTP/1.1" 200 -
127.0.0.1 - - [17/Mar/2025 01:22:06] "GET /activity-explaining.html HTTP/1.1" 200 -
----------------------------------------
Exception happened during processing of request from ('127.0.0.1', 53163)
Traceback (most recent call last):
  File "/opt/anaconda3/envs/myenv/lib/python3.8/socketserver.py", line 316, in _handle_request_noblock
    self.process_request(request, client_address)
  File "/opt/anaconda3/envs/myenv/lib/python3.8/socketserver.py", line 347, in process_request
    self.finish_request(request, client_address)
  File "/opt/anaconda3/envs/myenv/lib/python3.8/socketserver.py", line 360, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/opt/anaconda3/envs/myenv/lib/python3.8/http/server.py", line 664, in __init__
    super().__init__(*args, **kwargs)
  File "/opt/anaconda3/envs/myenv/lib/python3.8/socketserver.py", line 747, in __init__
    self.handle()
  File "/opt/anaconda3/envs/myenv/lib/python3.8/http/server.py", line 435, in handle
    self.handle_one_request()
  File "/opt/anaconda3/envs/myenv/lib/python3.8/http/server.py", line 423, in handle_one_request
    method()
  File "/opt/anaconda3/envs/myenv/lib/python3.8/http/server.py", line 671, in do_GET
    self.copyfile(f, self.wfile)
  File "/opt/anaconda3/envs/myenv/lib/python3.8/http/server.py", line 870, in copyfile
    shutil.copyfileobj(source, outputfile)
  File "/opt/anaconda3/envs/myenv/lib/python3.8/shutil.py", line 208, in copyfileobj
    fdst_write(buf)
  File "/opt/anaconda3/envs/myenv/lib/python3.8/socketserver.py", line 826, in write
    self._sock.sendall(b)
BrokenPipeError: [Errno 32] Broken pipe
----------------------------------------
127.0.0.1 - - [17/Mar/2025 01:22:06] code 404, message File not found
127.0.0.1 - - [17/Mar/2025 01:22:06] "GET /runestone/static/favicon.ico HTTP/1.1" 404 -
127.0.0.1 - - [17/Mar/2025 01:22:09] "GET /plan10.html HTTP/1.1" 200 -
127.0.0.1 - - [17/Mar/2025 01:22:09] "GET /_static/documentation_options.js HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:22:09] "GET /_static/language_data.js HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:22:09] "GET /_static/pygments.css HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:22:09] "GET /_static/basic.css HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:22:09] "GET /_static/bootstrap-sphinx.css HTTP/1.1" 304 -
127.0.0.1 - - [17/Mar/2025 01:22:09] "GET /activity-writing-1.html HTTP/1.1" 200 -
127.0.0.1 - - [17/Mar/2025 01:22:11] "GET /plan9.html HTTP/1.1" 200 -

Aborted!
